single.php

```diff
@@ -125,6 +125,8 @@
     ]
 );
 
+Context::set_p4_blocks_datalayer($context, $post);
```

Comment:
Added before the render to make sure that the `$post` has included the right content.